genesyscloud/resource_genesyscloud_routing_skill_group.go

```diff
@@ -201,13 +208,123 @@ func createOrUpdateSkillGroups(ctx context.Context, d *schema.ResourceData, meta
 		log.Printf("Updated skill group %s", skillGroupsRequest.Name)
 	}
 
+	// Update member division IDs
+	memberDivisionIds := d.Get("member_division_ids").([]interface{})
+	if memberDivisionIds == nil {
+		return readSkillGroups(ctx, d, meta)
+	}
+
+	log.Printf("Reading skill group %s member divisions", skillGroupsRequest.Name)
+
+	path = fmt.Sprintf("%s/api/v2/routing/skillgroups/%s/members/divisions", routingAPI.Configuration.BasePath, d.Id())
+	// GET /api/v2/routing/skillgroups/{skillGroupId}/members/divisions
+	apiSkillGroupMemberDivisionIds, diagErr := getAllSkillGroupMemberDivisionIds(d.Id(), path, routingAPI)
+	if err != nil {
+		return diagErr
+	}
+
+	log.Printf("Read skill group %s member divisions", skillGroupsRequest.Name)
+
+	schemaDivisionIds := interfaceListToStrings(memberDivisionIds)
+
+	toAdd, toRemove, diagErr := createListsForSkillgroupsMembersDivisionsPost(schemaDivisionIds, apiSkillGroupMemberDivisionIds, create, divisionId, meta)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	if len(toAdd) < 1 && len(toRemove) < 1 {
+		return readSkillGroups(ctx, d, meta)
+	}
+
+	log.Printf("Updating skill group %s member divisions", skillGroupsRequest.Name)
+
+	skillGroupsMemberDivisionIdsPayload := make(map[string][]string, 0)
+	if len(toRemove) > 0 {
+		skillGroupsMemberDivisionIdsPayload["removeDivisionIds"] = toRemove
+	}
+	if len(toAdd) > 0 {
+		skillGroupsMemberDivisionIdsPayload["addDivisionIds"] = toAdd
+	}
+
+	// POST /api/v2/routing/skillgroups/{skillGroupId}/members/divisions
+	response, err = apiClient.CallAPI(path, "POST", skillGroupsMemberDivisionIdsPayload, headerParams, nil, nil, "", nil)
+	if err != nil || response.Error != nil {
+		return diag.Errorf("Failed to create/update skill group %s member divisions: %s", skillGroupsRequest.Name, err)
+	}
+
+	log.Printf("Updated skill group %s member divisions", skillGroupsRequest.Name)
 	return readSkillGroups(ctx, d, meta)
 }
 
 func createSkillGroups(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	return createOrUpdateSkillGroups(ctx, d, meta, "/api/v2/routing/skillgroups", true)
 }
 
+func getAllAuthDivisionIds(meta interface{}) ([]string, diag.Diagnostics) {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	allIds := make([]string, 0)
+
+	divisionResourcesMap, err := getAllAuthDivisions(nil, sdkConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	for key, _ := range divisionResourcesMap {
+		allIds = append(allIds, key)
+	}
+
+	return allIds, nil
+}
+
+func createListsForSkillgroupsMembersDivisionsPost(schemaSkillGroupMemberDivisionIds []string, apiSkillGroupMemberDivisionIds []string,
+	create bool, divisionId string, meta interface{}) ([]string, []string, diag.Diagnostics) {
+	toAdd := make([]string, 0)
+	toRemove := make([]string, 0)
+	if stringInSlice("*", schemaSkillGroupMemberDivisionIds) {
+		if len(schemaSkillGroupMemberDivisionIds) > 1 {
```

Comment:
Should this code block be extracted into its own methods.  Also idiomatic go tries to avoid nested if..then functions